server/src/router/v1/index.ts

```diff
@@ -78,4 +79,65 @@ v1App.openapi(
 	},
 );
 
+// Majors
+const majorSchema = createSelectSchema(majors);
+
+v1App.openapi(
+	createRoute({
+		method: 'get',
+		path: '/majors',
+		tags: ['majors'],
+		summary: 'List all majors',
+		responses: {
+			[HttpStatusCodes.OK]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							majors: z.array(majorSchema),
+						}),
+					},
+				},
+				description: 'Successful response',
+			},
+		},
+	}),
+	async (c) => {
+		const foundMajors: Major[] = await db.select().from(majors);
+		return c.json({majors: foundMajors}, HttpStatusCodes.OK);
+	},
+);
+
+v1App.openapi(
+	createRoute({
+		method: 'get',
+		path: '/majors/{majorName}/users',
+		tags: ['majors'],
+		summary: 'List all users of a major',
+		middleware: [authMiddleWare('admin')],
+		responses: {
+			[HttpStatusCodes.OK]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							users: z.array(userSchema),
+						}),
+					},
+				},
+				description: 'Successful response',
+			},
+		},
+	}),
+	async (c) => {
+		const majorName  = c.req.param('majorName');
+
+		const foundUsers: User[] = await db.select().from(users).where(eq(users.major, majorName));
```

Comment:
may need to add try-catch
should also add check to see if `majorName` param is present